trunk/source/BuchiProgramProduct/src/de/uni_freiburg/informatik/ultimate/buchiprogramproduct/productgenerator/ProductGenerator.java

```diff
@@ -488,36 +497,11 @@ private void pruneNonProductSink(final BoogieIcfgLocation helper) {
 				generateNeverClaimAssumeStatement(new BooleanLiteral(null, BoogieType.TYPE_BOOL, true)));
 		mapNewEdge2OldEdge(seq, null);
 
-		// determine what kind of loop has to be added to this state based
-		// on the LTL NWA state of the predecessor
-		boolean added = false;
-		for (final String nwaState : mNWA.getStates()) {
-			for (final IcfgLocation node : predecessors) {
-				final BoogieIcfgLocation predecessor = (BoogieIcfgLocation) node;
-				if (!predecessor.getDebugIdentifier().toString().endsWith(nwaState)) {
-					continue;
-				}
-
-				// ok, the predecessor is from this node; now we add self loops to the helper state that keep us
-				// in this NWA state
-				for (final OutgoingInternalTransition<CodeBlock, String> autTrans : mNWA.internalSuccessors(nwaState)) {
-					if (autTrans.getSucc().equals(nwaState)) {
-						createNewStatementSequence(helper, seq, helper, autTrans.getLetter(), false);
-						added = true;
-					}
-				}
-
-				if (mNWA.isFinal(nwaState)) {
-					// and if the nwa state is accepting, this state will also be accepting
-					mAcceptingNodeAnnotation.annotate(helper);
-				}
-			}
-		}
 		// hacky shit: the ss is now useless; we remove it
-		if (added) {
-			seq.disconnectSource();
-			seq.disconnectTarget();
-		}
+		//		if (added) {
+		seq.disconnectSource();
+		seq.disconnectTarget();
+		//		}

```

Comment:
If you remove all the stuff (491 to 515) you dont need to create a self loop and remove it again ;)